ci-operator/step-registry/ibmcloud/provision/dedicatedhost/static/ibmcloud-provision-dedicatedhost-static-commands.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#pre create the dedicated hosts based on the ${SHARED_DIR}/dedicated_host.yaml and update "${SHARED_DIR}/dedicated_host.yaml
+#save the dedicated host name for master nodes in "${SHARED_DIR}/dedicated_host"
+#save the dedicated host name for worker nodes in "${SHARED_DIR}/dedicated_host" 
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+## create the dedicated host group and dedicated host, if succeed, save the dedicated host name to file"
+## input parameter: the resource group of the dhg, dh profile, zone, name of the dh and the file which save the created dh name.
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4 file=$5
+    local tmp family class dhg status
+    tmp=$(${IBMCLOUD_CLI} is dedicated-host-profile $dProfile --output JSON | jq -r '.family +","+ .class')
+    family=${tmp%,*}
+    class=${tmp#*,}
+    echo "family:" $family " class:" $class
+    dhg="${dhName}g"
+    run_command "${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg}"
+    
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
+    
+    status=$(waitingStatus ${dhName})
+    echo "${dhName} status: ${status}"
+    run_command "${IBMCLOUD_CLI} is dedicated-host ${dhName}"
+
+    if [[ "${status}" != "stable" ]]; then
+        echo "ERROR: fail to create the dedicated host with profile $dProfile on zone: $zone resouce group:$rg"
+        echo "delete the dh ${dhName}, which status is ${status} ..."
+        run_command "${IBMCLOUD_CLI} is dhd ${dhName} -f"
+        run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+    else
+        echo "${dhName}" > "${file}"        
+    fi
+}
+
+# waiting the pending dh status changed
+function waitingStatus() {
+    local dh=$1 status
+    COUNTER=0
+    while [ $COUNTER -lt 200 ]
+    do 
+        sleep 10
+        COUNTER=`expr $COUNTER + 20`
+        status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+        if [[ "${status}" != "pending" ]]; then
```

Comment:
What is the status when the DH is created successfully?

Reply:
expect stable status
> 02e7-97d3fafc-d54e-4c51-b061-9979f5d12250   ci-op-9z6l6hqw-e6ed7-dh   jp-tok-1   ci-op-9z6l6hqw-e6ed7-dhg   bx2-host-152x608   true                         stable            ci-op-9z6l6hqw-e6ed7-dhrg

Reply:
so `stable` is the expected status, right? Why run `break` once `stable` status is seen?

Reply:
when fail to create the dh, will fail to get "stable" status.

Reply:
The line 45 will check the status again, right? I am not sure how many types of status during DH creation, but sound break it once status get into non-pending, sounds not reasonable.

Reply:
If there is another status "pending"  -> "unknow" -> "stable", your codes look buggy.

Reply:
based on my test, it is "pending"-> "stable"/"failed"

Reply:
For such waitStatus function, the normal logic should be like run a loop to wait for an expected status, the result should be only two, one got the expected result, then return 0, another is timeout to get the expected result, then return non-zero value, the caller of the waitStatus function just need judge its return value, do not need a another check on `status` in caller functiona.